src/core/net/ip6.cpp

```diff
@@ -1230,6 +1231,22 @@ otError Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkM
             ExitNow(tunnel = true);
         }
 
+        if (aFromNcpHost && (nextHeader == kProtoTcp || nextHeader == kProtoUdp))
+        {
+            uint16_t dstPort;
+
+            // TCP/UDP shares header uint16_t srcPort, uint16_t dstPort
+            VerifyOrExit(aMessage.Read(aMessage.GetOffset() + sizeof(uint16_t), sizeof(dstPort), &dstPort) ==
+                             sizeof(dstPort),
+                         error = OT_ERROR_PARSE);
+            dstPort = HostSwap16(dstPort);
+            if (aMessage.IsLinkSecurityEnabled() && Get<ot::Ip6::Filter>().IsPortUnsecure(dstPort))
+            {
+                Get<ot::Ip6::Filter>().RemoveUnsecurePort(dstPort);
+                otLogInfoIp6("Removed unsecure port %d", dstPort);
```

Comment:
This practically changes and stricts the use of the unsecure port number:
- We can have only one node using unsecure port (during pairing) - since as soon as the node attaches and send traffic with security, we remove the unsecure port.
- We cannot use the same port number for communication to other already attached node (with security enabled).

If we go for this, I think we should ensure to update the documentation to mention the change.

Reply:
I  agree better document is required.